### PR TITLE
Processing data

### PR DESCRIPTION
**Pull request by abdozayan12**
In these project I Implement :

User model
A method that returns the 3 most recent posts for a given user.
A method that updates the posts counter for a user.
Post model
A method which returns the 5 most recent comments for a given post.
A method that updates the comments counter for a post.
A method that updates the likes counter for a post.

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -1,4 +1,12 @@
 class Comment < ApplicationRecord
   belongs_to :user
   belongs_to :post
```
Comment (mukhammadrizooff) [inline app/models/comment.rb]: - [ ] [REQUIRED] Good job so far! The associations between models should be set up correctly according, please implement `:author` correctly as below 🙏🏼

```ruby
belongs_to :author, class_name: 'User'
belongs_to :post
```

File: app/models/like.rb [hunk 2]
```diff
@@ -1,4 +1,12 @@
 class Like < ApplicationRecord
   belongs_to :user
   belongs_to :post
```
Comment (mukhammadrizooff) [inline app/models/like.rb]: - [ ] [REQUIRED] Good job so far! The associations between models should be set up correctly according, please implement `:author` correctly as below 🙏🏼

```ruby
belongs_to :author, class_name: 'User'
belongs_to :post
```

File: app/models/user.rb [hunk 3]
```diff
@@ -2,4 +2,8 @@ class User < ApplicationRecord
   has_many :posts, foreign_key: 'author_id'
   has_many :likes
   has_many :comments
```
Comment (mukhammadrizooff) [inline app/models/user.rb]: - [ ] [REQUIRED] Good job so far! The associations between models should be set up correctly according, please implement `foreign key` correctly as below 🙏🏼

```ruby
has_many :posts, foreign_key: 'author_id'
has_many :comments, foreign_key: 'author_id'
has_many :likes, foreign_key: 'author_id'
```

**Review by mukhammadrizooff — CHANGES_REQUESTED** [review]
Hi @abdozayan12, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @abdozayan12,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**